File: src/main/java/com/company/erp/features/devices/dto/RegisterDeviceRequest.java
package com.company.erp.features.devices.dto;

import com.company.erp.features.devices.entity.DevicePlatform;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record RegisterDeviceRequest(
        @NotBlank @Size(max = 128) String deviceId,
        @NotBlank                   String fcmToken,
        @NotNull  DevicePlatform    platform,
        @Size(max = 32)             String appVersion
) {
}

File: app/Features/Devices/Requests/RegisterDeviceRequest.php
<?php

namespace App\Features\Devices\Requests;

use App\Features\Devices\Models\DevicePlatform;
use App\Support\Http\ApiFormRequest;
use Illuminate\Validation\Rule;

class RegisterDeviceRequest extends ApiFormRequest
{
    public function rules(): array
    {
        return [
            'deviceId' => ['required', 'string', 'max:128'],
            'fcmToken' => ['required', 'string'],
            'platform' => ['required', Rule::in(DevicePlatform::values())],
            'appVersion' => ['nullable', 'string', 'max:32'],
        ];
    }
}
